### DISCUSSION TITLE
Can KitchenSink app features be integrated into the MindMap app?

### DISCUSSION THREAD

**Discussion by QuangLanh** [clientIO/joint#2757] [Q&A]
### Introduction

Currently, I want to integrate all the features of KitchenSink app into MindMap app so that I can use all the features of KitchenSink for MindMap UI as well. Is that possible?
![Thiết kế chưa có tên](https://github.com/user-attachments/assets/93681b15-8652-4a64-af2d-c21fe224a304)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
It is possible, but not every feature in the KitchenSink is applicable in a mind mapping application e.g snaplines, gridlines, send to back make no sense in the context of a mind map.
- You can add `Stencil` to your mind map application and integrate it with the `TreeLayoutView` (you don't want the elements to be added freely to the graph, but to be attached to the tree). Example to be found in the [docs](https://docs.jointjs.com/learn/features/drag-and-drop/tree-layout-view).
- `Toolbar` is already in there. So you just need to add your own tools / buttons.
- You can add the `Inspector` to edit the elements properties too.

**Reply by QuangLanh** [reply]
Currently, I have three questions I want to ask you:

First, why can't I display the images on the stencil part?
![image](https://github.com/user-attachments/assets/7089696e-f0ad-4ede-85e0-232aea77b64b)
![image](https://github.com/user-attachments/assets/fd4aa40c-598e-498d-a943-6d64c3892ecd)

Second, although I can drag the image into the paper, I can't actually drop it into the paper.
![image](https://github.com/user-attachments/assets/4d6a3fbd-81d5-4b1e-a346-806c82316492)

Third, when I drag the elements on the paper, they don't move at all.

**Reply by kumilingus** [reply]
1. The `Idea` shape is a custom shape with custom view. The JSON of `Idea` shape looks more like this (there are no `attrs`):
```ts
  {
      type: 'Idea',
      label: 'My Label',
      annotations: [],
      image: '',
      imageSize: 50,
      spacing: 5,
      placeholder: ' ',
      outlineColor: 'black',
      fillColor: 'white',
      line: true            
  }
  ```

2. Here's an example how to integrate the `Stencil` plugin with `TreeLayoutView`: https://docs.jointjs.com/learn/features/drag-and-drop/tree-layout-view/

3. They do not move, because the example is using the `ui.TreeLayouView` plugin that is preventing them from doing so / and the paper is set to `interactive: false`.
